packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -0,0 +1,355 @@
+    );
+
+    this.hbarLimiter.addExpense(cost, Date.now(), requestId);
+    this.captureMetrics(
+      SDKClient.transactionMode,
+      transactionType,
+      Status.Success,
+      cost,
+      gasUsed,
+      callerName,
+      interactingEntity,
+    );
+  };
+
+  /**
+   * Retrieves the cost metric for consensus node client operations.
+   *
+   * @returns {Histogram} - The histogram metric tracking the cost of consensus node client operations.
+   */
+  public getCostMetric(): Histogram {
+    return this.consensusNodeClientHistogramCost;
+  }
+
+  /**
+   * Retrieves the gas fee metric for consensus node client operations.
+   *
+   * @returns {Histogram} - The histogram metric tracking the gas fees of consensus node client operations.
+   */
+  public getGasFeeMetric(): Histogram {
+    return this.consensusNodeClientHistogramGasFee;
+  }
+
+  /**
+   * Initialize consensus node cost metrics
+   * @param {Registry} register
+   * @returns {Histogram} Consensus node cost metric
+   */
+  private initCostMetric(register: Registry): Histogram {
+    const metricHistogramCost = 'rpc_relay_consensusnode_response';
+    register.removeSingleMetric(metricHistogramCost);
+    return new Histogram({
+      name: metricHistogramCost,
+      help: 'Relay consensusnode mode type status cost histogram',
+      labelNames: ['mode', 'type', 'status', 'caller', 'interactingEntity'],
+      registers: [register],
+    });
+  }
+
+  /**
+   * Initialize consensus node gas metrics
+   * @param {Registry} register
+   * @returns {Histogram} Consensus node gas metric
+   */
+  private initGasMetric(register: Registry): Histogram {
+    const metricHistogramGasFee = 'rpc_relay_consensusnode_gasfee';
+    register.removeSingleMetric(metricHistogramGasFee);
+    return new Histogram({
+      name: metricHistogramGasFee,
+      help: 'Relay consensusnode mode type status gas fee histogram',
+      labelNames: ['mode', 'type', 'status', 'caller', 'interactingEntity'],
+      registers: [register],
+    });
+  }
+
+  /**
+   * Captures and records metrics for a transaction.
+   * @private
+   * @param {string} mode - The mode of the transaction (e.g., consensus mode).
+   * @param {string} type - The type of the transaction.
+   * @param {Status} status - The status of the transaction.
+   * @param {number} cost - The cost of the transaction in tinybars.
+   * @param {any} gas - The gas used by the transaction.
+   * @param {string} caller - The name of the caller executing the transaction.
+   * @param {string} interactingEntity - The entity interacting with the transaction.
+   * @returns {void}
+   */
+  private captureMetrics = (
+    mode: string,
+    type: string,
+    status: Status,
+    cost: number,
+    gas: any,
+    caller: string,
+    interactingEntity: string,
+  ): void => {
+    const resolvedCost = cost ? cost : 0;
+    const resolvedGas = gas ? (typeof gas === 'object' ? gas.toInt() : gas) : 0;
```

Comment:
Why are we doing this `typeof gas === 'object'` check here?

This is an example of a line which should be commented with a comment explaining "why".

Reply:
Ah yeah that was part of the old implementation, but with the new update, gas is strictly typed as a number, so this is no longer needed. Great catch! I've updated it with the fix.

Reply:
Ah yeah that was part of the old implementation, but with the new update, gas is strictly typed as a number, so this is no longer needed. Great catch! I've updated it with the fix.